pkg/workceptor/remote_work_test.go

```diff
@@ -220,65 +258,208 @@ func TestRemoteWorkLifecycleOperations(t *testing.T) {
-			mockBaseWorkUnit.EXPECT().GetStatusWithoutExtraData().Return(&workceptor.StatusFileData{}).AnyTimes()
+			mockBaseWorkUnit.EXPECT().GetStatusWithoutExtraData().DoAndReturn(func() *workceptor.StatusFileData {
+				return &workceptor.StatusFileData{}
+			}).AnyTimes()
 			mockBaseWorkUnit.EXPECT().GetStatusCopy().Return(workceptor.StatusFileData{
 				ExtraData: remoteExtraData,
 			}).AnyTimes()
-			mockBaseWorkUnit.EXPECT().GetWorkceptor().Return(&workceptor.Workceptor{}).AnyTimes()
+			mockBaseWorkUnit.EXPECT().GetWorkceptor().Return(w).AnyTimes()
 
 			if tt.operation == "cancel" || tt.operation == "release" {
-				if !tt.remoteStarted {
-					mockBaseWorkUnit.EXPECT().UpdateFullStatus(gomock.Any()).Do(func(updateFunc interface{}) {
-						updateFunc.(func(*workceptor.StatusFileData))(&workceptor.StatusFileData{
-							ExtraData: remoteExtraData,
-						})
+				mockBaseWorkUnit.EXPECT().UpdateFullStatus(gomock.Any()).Do(func(updateFunc interface{}) {
+					updateFunc.(func(*workceptor.StatusFileData))(&workceptor.StatusFileData{
+						ExtraData: remoteExtraData,
 					})
-					if tt.operation == "cancel" {
-						mockBaseWorkUnit.EXPECT().UpdateBasicStatus(workceptor.WorkStateFailed, "Locally Cancelled", int64(0))
-					} else {
-						mockBaseWorkUnit.EXPECT().Release(true).Return(nil)
-					}
-				}
+				})
 			}
 
 			var err error
-			switch tt.operation {
-			case "start":
+			switch tt.name {
+			case "start not started":
+				mockBaseWorkUnit.EXPECT().ID().Return("test-id").AnyTimes()
+				messages := []string{
+					"execution\n", // Hello message with remote node ID
+					"Work unit created with ID execution. Send stdin data and EOF.\n",
+					"OK\n", // Acknowledgment after stdin sent
+				}
+				mockConn, _ := createRemoteWorkNetworkSetup(t, ctrl, contextWithCancel, messages)
+				mockNetceptor.EXPECT().GetClientTLSConfig(gomock.Any(), gomock.Any(), gomock.Any()).Return(&tls.Config{}, nil).AnyTimes()
+				mockNetceptor.EXPECT().DialContext(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mockConn, nil).AnyTimes()
+				mockBaseWorkUnit.EXPECT().Load().Return(nil).AnyTimes()
+				mockBaseWorkUnit.EXPECT().StdoutFileName().Return("/tmp/start_not_started/stdout").AnyTimes()
+				mockBaseWorkUnit.EXPECT().UpdateFullStatus(gomock.Any()).Do(func(updateFunc interface{}) {
+					updateFunc.(func(*workceptor.StatusFileData))(&workceptor.StatusFileData{
+						ExtraData: remoteExtraData,
+					})
+				}).AnyTimes()
+				mockBaseWorkUnit.EXPECT().UnitDir().Return("/tmp/start_not_started").AnyTimes()
+				// Add expectation for UpdateBasicStatus and LastUpdateError which will be called by the monitoring goroutine
+				mockBaseWorkUnit.EXPECT().UpdateBasicStatus(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+				mockBaseWorkUnit.EXPECT().LastUpdateError().Return(nil).AnyTimes()
+				// Create temporary file for stdin
+				_ = os.MkdirAll("/tmp/start_not_started", 0o755)
```

Comment:
Pablo, the use of /tmp files is common across our codebase and causes flakiness, better to use t.TempDir() 

```suggestion
// Correct approach:
tempDir := t.TempDir() // Automatically cleaned up after test
stdinPath := filepath.Join(tempDir, "stdin")
if err := os.WriteFile(stdinPath, []byte{}, 0755); err != nil {
    t.Fatalf("Failed to create stdin: %v", err)
}
mockBaseWorkUnit.EXPECT().UnitDir().Return(tempDir).AnyTimes()
```

Reply:
I cannot use `TempDir()` here because `stdin` is hardcoded in the code. If I use `TempDir()` it will generate something like `stdin0u401ummd` and the code will complain that it cannot find a `stdin` file.

Reply:
After moving this to a helper function the linter is not complaining anymore either hehe